Source/controls/touch/renderers.h

```diff
@@ -0,0 +1,156 @@
+#pragma once
+
+#if defined(VIRTUAL_GAMEPAD) && !defined(USE_SDL1)
+
+#include "controls/touch/gamepad.h"
+#include "engine/surface.hpp"
+#include "utils/png.h"
+#include "utils/sdl_ptrs.h"
+
+namespace devilution {
+
+class VirtualDirectionPadRenderer {
+public:
+	VirtualDirectionPadRenderer(VirtualDirectionPad *virtualDirectionPad)
+	    : virtualDirectionPad(virtualDirectionPad)
+	    , padSurface(nullptr)
+	    , knobSurface(nullptr)
+	{
+	}
+
+	void Render(const Surface &out);
+
+private:
+	VirtualDirectionPad *virtualDirectionPad;
+	SDLSurfaceUniquePtr padSurface;
+	SDLSurfaceUniquePtr knobSurface;
+
+	void RenderPad(const Surface &out);
+	void RenderKnob(const Surface &out);
+};
+
+class VirtualPadButtonRenderer {
+public:
+	VirtualPadButtonRenderer(VirtualPadButton *virtualPadButton)
+	    : virtualPadButton(virtualPadButton)
+	    , attackSurface(nullptr)
+	    , pressedAttackSurface(nullptr)
+	    , talkSurface(nullptr)
+	    , pressedTalkSurface(nullptr)
+	    , itemSurface(nullptr)
+	    , pressedItemSurface(nullptr)
+	    , objectSurface(nullptr)
+	    , pressedObjectSurface(nullptr)
+	    , castSurface(nullptr)
+	    , pressedCastSurface(nullptr)
+	    , cancelSurface(nullptr)
+	    , pressedCancelSurface(nullptr)
+	    , blankSurface(nullptr)
+	    , pressedBlankSurface(nullptr)

```

Comment:
I thought I took these out. They're completely unnecessary.
```suggestion
```

Reply:
As long as you don't rebase I think it will be fine :)